pkg/archive.py

```diff
@@ -186,15 +187,16 @@ def add_dir(self,
     """
     if not (name == self.root_directory or name.startswith('/') or
             name.startswith(self.root_directory + '/')):
-      name = os.path.join(self.root_directory, name)
+      name = self.root_directory + '/' + name
```

Comment:
I think keeping the / visible helps mark that we are handling these values two different ways.  This way a future maintainer will be less inclined to see  replace   name = runfiles_join(root_dir, name) with os.path.join "for portability".
I'll keep this for now.

OTOH,  archive.py is a mess and could use a rewrite for clarity. At that time, maybe we could define a Path type to hold these values, with an append() operation and accessors to get the os correct path.

Reply:
`self.root_directory + '/' + name` is not equivalent to `os.path.join(self.root_directory, name)` if root directory is an empty string.
An empty root_directory, as opposed to ".", is desirable because it avoids creating a "." entry in the tar archive (from line 288 below). And a "." entry is not really a good idea -- it will cause a very natural command like `tar xf foo.tar -C /tmp` to exit with an error like:
```
tar: .: Cannot utime: Operation not permitted
tar: .: Cannot change mode to rwxr-xr-t: Operation not permitted
```
since tar will attempt to update the time of "." aka the "/tmp", and that obviously does not work.

Reply:
That sounds like you are proposing a new feature or a behavior change. Something like: "Add the ability to archive file paths without leading './' or '/'"

That seems like a reasonable feature, but let's talk about that outside the context of this PR.  If you would care to open an issue that would be fine.

Reply:
Isn't this https://github.com/bazelbuild/rules_pkg/issues/50, or something like it?